_ml-commons-plugin/api/mcp-server-apis/register-mcp-tools.md

```diff
@@ -0,0 +1,123 @@
+`parameters` | Object | Optional | The parameters for this tool. The parameters are highly dependent on the tool type. You can find information about specific tool types in [Tools]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/tools/index/).
+`attributes` | Object | Optional | The attributes for this tool. The most critical property in this field is the tool's schema with key `input_schema`, tool's schema will be sent to llm so llm can figure out the correct format parameters to execute the tool.
+
+#### Example request: WebSearchTool
+
+```json
+POST /_plugins/_ml/mcp/tools/_register
+{
+    "tools": [
+        {
+            "type": "WebSearchTool",
+            "name": "GoogleSearchTool",
+            "attributes": {
+                "input_schema": {
+                    "type": "object",
+                    "properties": {
+                        "engine": {
+                            "type": "string",
+                            "description": "search engine will be used by tool"
+                        },
+                        "query": {
+                            "type": "string",
+                            "description": "search query parameter, will be used perform search by engine"
+                        },
+                        "next_page": {
+                            "type": "string",
+                            "description": "search result's next page link, if this is passed, the WebSearchTool will fetch the next_page result with the link and crawl the links on the page"
+                        }
+                    },
+                    "required": [
+                        "engine", "query"
+                    ]
+                },
+                "strict": false
+            }
+        }
+    ]
+}
+```
+{% include copy-curl.html %}
+
+#### Example request: PPLTool
+
+```json
+POST /_plugins/_ml/agents/_register
+{
+    "type": "PPLTool",
+    "name": "TransferQuestionToPPLAndExecuteTool",
+    "description": "Use this tool to transfer natural language to generate PPL and execute PPL to query inside. Use this tool after you know the index name, otherwise, call IndexRoutingTool first. The input parameters are: {index:IndexName, question:UserQuestion}",
+    "parameters": {
+        "model_id": "${your_model_id}",
+        "model_type": "FINETUNE"
+    },
+    "attributes":{
+        "input_schema":{
+            "type":"object",
+            "properties":{
+                "question": {
+                    "type": "string",
+                    "description": "The user natual language question that needs to be transferred to PPL"
+                },
+                "index": {
+                    "type": "string",
+                    "description": "The index that the PPL will be generated based on"
+                }
+            }
+        }
+    }
+}
+```
+{% include copy-curl.html %}
+
+#### Example response
+
+OpenSearch responds with an created result in each instance:
```

Comment:
Are these the tool IDs and the status of the creation for each tool?

Reply:
These are instance ids and we don't have status for each single tool, the `created` means all tools in the request are been created on the instance.

Reply:
@zane-neo Could we clarify this? An instance of what?

Reply:
@kolchfa-aws An instance usually mean a node in the cluster.